src/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs

```diff
@@ -84,7 +84,11 @@ private void CloseCore()
         {
             // Make sure that we configure the wait state holder for this process object, which we can only do once we have a process ID.
             Debug.Assert(_haveProcessId, $"{nameof(ConfigureAfterProcessIdSet)} should only be called once a process ID is set");
-            GetWaitState(); // lazily initializes the wait state
+            // Initialize WaitStateHolder for non-child processes
+            if (_waitStateHolder == null)
+            {
+                _waitStateHolder = new ProcessWaitState.Holder(_processId);
+            }
```

Comment:
We're adding non-trivial expense here to cases that never previously had it.  Is it really worthwhile?

Reply:
I found it simpler to initialize this once than to figure out the lazy initialization doesn't cause issues. I'll revert this.